src/utils/entity-utils.ts

```diff
@@ -11,10 +12,15 @@ export const matchEntitiesByParams = (params: Record<string, string>, paramMatch
 };
 
 export const matchEntitiesByBodyFilters = (body: Request['body'], bodyMatch: Optional<Record<string, string>>) => {
-  return (entity: any) => bodyMatch ?  Object.entries(bodyMatch).every(([bodyKey, entityKey]) => {
+  return (entity: any) => bodyMatch ? Object.entries(bodyMatch).every(([bodyKeyPath, entityKeyPath]) => {
+    if (entityKeyPath === IGNORE_KEY) {
+      return true;
+    }
+    const entityValue = getObjValue(entity, entityKeyPath);
+    const value = getObjValue(body, bodyKeyPath);
 
     // todo: we need to handle isArray(value) too
-    return entityKey === IGNORE_KEY || entity[entityKey] === body[bodyKey];
+    return entityValue === value;
```

Comment:
I think we need to do this for the queryParams and paramsMatch as well but for values only.